caddyhttp/proxy/reverseproxy.go

```diff
@@ -231,6 +232,26 @@ func (rp *ReverseProxy) UseInsecureTransport() {
 	}
 }
 
+func (rp *ReverseProxy) SetTransportCARoots(certPool *x509.CertPool) {
+	if rp.Transport == nil {
+		transport := &http.Transport{
+			Proxy:               http.ProxyFromEnvironment,
+			Dial:                defaultDialer.Dial,
+			TLSHandshakeTimeout: 10 * time.Second,
+			TLSClientConfig:     &tls.Config{RootCAs: certPool},
+		}
```

Comment:
Makes sense. Something like this?

```go
func (rp *ReverseProxy) UseInsecureTransport() {
	rp.createHttp();
	rp.Transport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
}

func (rp *ReverseProxy) SetTransportCARoots(certPool *x509.CertPool) {
	rp.createHttp();
	rp.Transport.(*http.Transport).TLSClientConfig = &tls.Config{RootCAs: certPool}
}

func (rp *ReverseProxy) createHttp() {
	if rp.Transport == nil {
		transport := &http.Transport{
			Proxy:               http.ProxyFromEnvironment,
			Dial:                defaultDialer.Dial,
			TLSHandshakeTimeout: 10 * time.Second,
		}
		if httpserver.HTTP2 {
			http2.ConfigureTransport(transport)
		}
		rp.Transport = transport
	}
}
```

I've removed the

```go
if transport, ok := rp.Transport.(*http.Transport); ok {
```
block, since the `rp.Transport.(*http.Transport)` is now always created, is that right?

I think there is one more place where the function will be called. I'll replace it, if you find this correct.